realm/realm-library/src/main/cpp/io_realm_internal_OsObject.cpp

```diff
@@ -162,18 +162,22 @@ static void finalize_object(jlong ptr)
 
 static inline size_t do_create_row(jlong shared_realm_ptr, jlong table_ptr)
 {
-    auto& shared_realm = *(reinterpret_cast<SharedRealm*>(shared_realm_ptr));
     auto& table = *(reinterpret_cast<realm::Table*>(table_ptr));
-    shared_realm->verify_in_write();
+    if (shared_realm_ptr) {
```

Comment:
`SharedRealm.nativePtr` is final, and `close()` won't free the memory. `shared_realm_ptr` should never be `null` i think. This check is redundant.